src/plugins/graph/imports/components/axis/components/axis-or-legend-attribute-menu.tsx

```diff
@@ -70,14 +70,19 @@ const _AxisOrLegendAttributeMenu = ({ place, target, portal,
     });
   }, [attribute?.name, data?.attributes, dataConfig, labelText, setLabelText, attrId]);
 
+  const updateTargetMenuClasses = (isOpen: boolean) => {
+    target?.classList.toggle("target-open", isOpen);
+    target?.classList.toggle("target-closed", !isOpen);
+  };
+
```

Comment:
This is an interesting approach.  It is a bit confusing for one component to be changing the classes of some other component. This might violate some React virtual DOM principle. 

I don't know if it would work, but could this menu component take a property: `onOpenOrClose` which is passed the `isOpen` boolean. Then the component which is using this menu object can implement its own class switching by rerendering.

Reply:
> Then the component which is using this menu object can implement its own class switching by rerendering.

I would like to do it that way, but I am trying to figure out what I have done wrong in react land.  When I try to update a piece of state from the `onOpenClose`, it gets mad the first time. The only way I get it working without error is what I have just pushed, which moves the same direct dom manipulation as before to the same file and operates on the ref.current:  ([commit](https://github.com/concord-consortium/collaborative-learning/pull/1883/commits/6cf53e285b032b948e96c2b7db7ac01178de9914)) I know there is a proper way to do it.  Perhaps we can plan to go over it in story hour if I don't see it more clearly in the morning.